### PR TITLE
Switch to Maplibre GL JS and generate swipe map assets using input GeoTIFFs

### PR DESCRIPTION
**Pull request by rudokemper**
Closes https://github.com/ConservationMetrics/guardianconnector-change-detection/issues/7.

**High level summary:** I adapted GCCD to accept optional before (`t0`) and after (`t1`) GeoTIFF inputs, and to generate swipe map assets if provided. In so doing, I replaced Mapbox GL JS with MapLibre GL JS, because the latter library has support for the `pmtiles` tile format which I am leveraging to visualize the before/after scenes. There are also real world fixtures in the repo now, representing the actual data that GCCD will be working with at first.

**Why MapLibre and PMTiles**: [PMTiles](https://github.com/protomaps/PMTiles) is a new cloud-optimized single-file tile format for vector and raster data. Much like mbtiles, pmtiles offers the advantage over the `xyz/` directory format that all map tiles are stored and can be deduped from a single file. Pmtiles offers an additional advantage over mbtiles (a SQLite database): they can be rendered using JS in the browser without needing an additional server (like Tileserver-GL) to serve the tiles. The pmtiles Javascript library has a plugin for MapLibre GL that uses its `addProtocol` feature.

An additional motivation for substituting Mapbox for MapLibre is that we don't need to worry about access tokens (and potential costs if we exceed the Mapbox freemium API threshold) anymore, and we are staying closer to our FOSS principles.

**Detailed description:**

Granular level of detail because I realize there is a lot going on here! (Each commit is scoped to a particular piece so when I merge, I won't squash.)

Main script
* `main.py` now accepting arguments with GeoTIFFs file paths (and rage quits if only one has been provided).
* MapLibre GL JS implemented across the map templates. 
  * We are now using Bing satellite from Virtual Earth as the background map, and using fonts and glyphs from Protomaps.
  * MAPBOX_ACCESS_TOKEN and MAPBOX_STYLE env vars removed from the scripts.
  * Created a new template for map swipe, implementing the `pmtiles` protocol to visualize the before/after maps; for now, using self-hosted assets from [maplibre-gl-compare](https://github.com/maplibre/maplibre-gl-compare) since they are not (yet) hosted anywhere for public usage.
  * Minor thing - adopted a `mapgl` naming convention since a lot of our outputs are map library agnostic.
* In `utils.py`, `copy_geojson_file` renamed to `copy_input_files` and expanded to also copy over the GeoTIFFs.
* In `generate_maps.py`, `generate_html_map` is now generating either the existing HTML map template or the new swipe one, depending on whether GeoTIFF arguments were provided.
* In `generate_tiles.py`:
  * `generate_pmtiles_from_geotiff` is running through a series of steps: `tif` -> `xyz/` -> `mbtiles` -> `pmtiles`. I know, seems redundant, but I couldn't get a more direct conversion running as of yet. I'll revisit this in the future; I think there will be a lot more support for working with pmtiles soon on the horizon.
  * I turned each of these steps into more bite-size functions, and adapted the rest of the script to use these too, where needed.
  * To accomplish these conversions, the use of `go-pmtiles` and `gdal_translate` libraries was needed. I documented this in the READMEs and also added these to the Dockerfiles.

Docker services 
* `docker-generator.py` adapted with the same script changes.
* All Dockerfiles (generate and FastAPI) are pulling in the required libraries that have been added (GDAL and pmtiles).
* `run.sh` and `docker-compose.yml` updated to optionally accept the new flags, and run a docker command with or without them.

Other
* `example.geojson` updated with real data; `example_t0` + `example_t1` GeoTIFFs added.
* READMEs updated and sample swipe map screenshot added.

**Disclosure:**

While I have tested the script with the Docker Compose and Python Package execution methods, I have not yet done so with the FastAPI service endpoint. I will turn to that next as I work on the frizzle component for handling change detection messages.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,27 +13,26 @@ The script follows these steps:
...
-3. Generate HTML Mapbox map for previewing change detection alert
-4. Generate vector MBTiles from GeoJSON
-5. Generate raster XYZ tiles from satellite imagery and bounding box
-6. Convert raster XYZ directory to MBTiles
-7. Generate stylesheet with both raster and vector MBTiles files
-8. Download and copy over fonts and glyphs (needed for map style)
-9. Generate overlay map HTML map (to preview style)
-10. Serve maps using `tileserver-gl`
-11. Generate composite MBTiles from `tileserver-gl` map loading the style
+3. Generate Generate PMTiles for GeoTIFFS (if provided)
```
Comment (IamJeffG) [inline README.md:16]:
```suggestion
3. Generate PMTiles from GeoTIFFs (if provided)
```

File: httpservice/Dockerfile [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
+RUN git clone https://github.com/felt/tippecanoe.git && cd tippecanoe && make -j && make install && cd .. && rm -r tippecanoe
+RUN apt-get install -y binutils libproj-dev gdal-bin
+RUN apt-get install -y wget && \
+    wget https://github.com/protomaps/go-pmtiles/releases/download/v1.11.0/go-pmtiles_1.11.0_Linux_x86_64.tar.gz && \
+    tar -xzf go-pmtiles_1.11.0_Linux_x86_64.tar.gz && \
+    rm go-pmtiles_1.11.0_Linux_x86_64.tar.gz && \
+    mv pmtiles /usr/local/bin
 
 #-#-#-#-#-#-#
 FROM python:3.11-slim
```
Comment (IamJeffG) [inline httpservice/Dockerfile:13]:
```suggestion
FROM python:3.11-slim
COPY --from=tippecanoe-build-image /usr/local/bin/tile-join /usr/local/bin/tippecanoe* /usr/local/pmtiles /usr/local/bin/

```

File: httpservice/Dockerfile [hunk 3]
```diff
@@ -1,8 +1,13 @@
 FROM python:3.11-slim as tippecanoe-build-image
 
 RUN apt-get update && apt-get install -y --no-install-recommends build-essential libsqlite3-dev zlib1g-dev git
-RUN git clone https://github.com/mapbox/tippecanoe.git && cd tippecanoe && make -j && make install && cd .. && rm -r tippecanoe
-
+RUN git clone https://github.com/felt/tippecanoe.git && cd tippecanoe && make -j && make install && cd .. && rm -r tippecanoe
+RUN apt-get install -y binutils libproj-dev gdal-bin
```
Comment (IamJeffG) [inline httpservice/Dockerfile:5]: This line may need to go in the second image below, and not here.

The image we are in now is just a temporary one that we're going to scavenge binaries out of in the image we'll actually deploy below.  I imagine that one we deploy needs to have gdal.

**Review by IamJeffG — COMMENTED** [review]
I did not look too closely but offered some suggestions.

One trick with the HTTP server is that it has no way to know if something went wrong, so it always returns successfully (but without having written files where they need to go). This is because the gccd library catches ("swallows") all exceptions (sys.exit).  I think in a future PR we will want to have it actually fail loudly, surfacing any errors up to the top to let the caller (you, or the Http server, for example) know.  We can talk about the philosophy of error handling at some point.

File: httpservice/Dockerfile (hunk 3 above)
Comment (rudokemper) [reply httpservice/Dockerfile:5]: Ah, yeah. Oversight; thanks.

**Comment by rudokemper** [comment]
The suggestions were implemented, and the observation about improving our error handling for the HTTP server were captured in the engineering tasks table.

**Review by IamJeffG — APPROVED** [review]

**Pull request merged**